src/main/kotlin/app/batch/HBaseWriter.kt

```diff
@@ -168,6 +175,23 @@ class HBaseWriter : ItemWriter<DecompressedStream> {
         }
     }
 
+    fun idObject(json: JsonObject): JsonObject? {
+        val id = json.get("_id")
+
+        if (id.isJsonObject) {
+            return id.asJsonObject
+        }
+        else if (id.isJsonPrimitive) {
+            val value = id.asJsonPrimitive
+            val asObject = JsonObject()
+            asObject.addProperty("id", value.asString)
```

Comment:
This should be "_id".

Reply:
This needs to be a constant and re-used everywhere please, not a magic string :-)